docs/apis-tools/tasklist-api-rest/tasklist-api-rest-authentication.md

```diff
@@ -5,63 +5,137 @@ sidebar_position: 2
-| authorization server url | Token issuer server                             | -                     |
-
-Send a token issue _POST_ request to the authorization server with the following content:
-
-```json
-{
-  "client_id": "<client-id>",
-  "client_secret": "<client-secret>",
-  "audience": "<audience>",
-  "grant_type": "client_credentials"
-}
-```
-
-Refer to the following example with _curl_:
+import Tabs from "@theme/Tabs";
+import TabItem from "@theme/TabItem";
+
+All Tasklist API requests require authentication. To authenticate, generate a JWT token and pass it in each request.
+
+## Generating a token
+
+<Tabs groupId="authentication" defaultValue="saas" queryString values={
+[
+{label: 'SaaS', value: 'saas' },
+{label: 'Self-Managed', value: 'self-managed' },
+]}>
+<TabItem value='saas'>
+
+1. [Create client credentials](/guides/setup-client-connection-credentials.md) in the **Clusters > Cluster name > API** tab of [Camunda Console](https://console.camunda.io/).
+2. Add permissions to this client for **Tasklist**.
+3. Upon creating the client, capture the following values required to generate a token:
+   <!-- this comment convinces the markdown processor to still treat the table as a table, but without adding surrounding paragraphs. 🤷 -->
+   | Name                     | Environment variable name        | Default value                                |
+   | ------------------------ | -------------------------------- | -------------------------------------------- |
+   | Client ID                | `ZEEBE_CLIENT_ID`                | -                                            |
+   | Client Secret            | `ZEEBE_CLIENT_SECRET`            | -                                            |
+   | Authorization Server URL | `ZEEBE_AUTHORIZATION_SERVER_URL` | `https://login.cloud.camunda.io/oauth/token` |
+   | Tasklist REST Address    | `CAMUNDA_TASKLIST_BASE_URL`      | -                                            |
+   <!-- this comment convinces the markdown processor to still treat the table as a table, but without adding surrounding paragraphs. 🤷 -->
+   :::tip
+   When client credentials are created, the `Client Secret` is only shown once. Save this `Client Secret` somewhere safe.
+   :::
+4. Execute an authentication request to the token issuer:
+   ```bash
+   curl --request POST ${ZEEBE_AUTHORIZATION_SERVER_URL} \
+       --header 'Content-Type: application/x-www-form-urlencoded' \
+       --data-urlencode 'grant_type=client_credentials' \
+       --data-urlencode 'audience=tasklist.camunda.io' \
+       --data-urlencode "client_id=${ZEEBE_CLIENT_ID}" \
+       --data-urlencode "client_secret=${ZEEBE_CLIENT_SECRET}"
+   ```
+5. A successful authentication response looks like the following:
+   ```json
+   {
+     "access_token": "<TOKEN>",
+     "expires_in": 300,
+     "refresh_expires_in": 0,
+     "token_type": "Bearer",
+     "not-before-policy": 0
+   }
```

Comment:
Do we need to do anything about the difference in the responses? The expiration immediately caught my eye.

Reply:
This specific output comes from a self-managed environment. In a SaaS environment, the only notable difference is the absence of a `refresh_expires_in` property. The default `expires_in` differs across environments, but in a SaaS environment that value can be just about anything, because the tokens are cached on the server rather than re-generated on each auth request. 

So a SaaS user would see an extra field in this sample response compared to reality. Do you think that's worse than a SM user seeing an absence of a field in this sample response compared to reality? I don't personally think either is very confusing, so I'd avoid showing a different sample response based on environment, to avoid unnecessary complexity.

Reply:
Oh, and now I see from your next comment that you were referring to the _value_ of the `expires_in`, not the presence of the `refresh_expires_in`. 😅 😬